R/ModelFit-class.R

```diff
@@ -0,0 +1,99 @@
+
+# constructor
+#' Constructs a new ModelFit object
+#'
+#' @param par list of named parameters
+#' @param distribution character name of distribution
+#' @param metric character name of metric
+#' @param value numeric result of applying metric to observed data and fitted distribution
+#' @param density_function function which takes a vector of coordinates, a list of parameters and
+#' a logical determining whether the output should be scaled; returns the fitted distribution density
+#' @param seg_start integer histogram index start of uniform segment
+#' @param seg_end integer histogram index end of uniform segment
+#' @param ... additional parameters, allowing child classes to be built
+#'
+#' @return a Histogram object
+new_ModelFit <- function(
+    par = NULL,
+    distribution = NULL,
+    metric = NULL,
+    value = NULL,
+    density_function = NULL,
+    seg_start = NULL,
+    seg_end = NULL,
+    ...
+  ){
+
+  # Checking types
+  stopifnot(is.list(par) || is.null(par))
+  stopifnot(is.character(distribution))
+  stopifnot(is.character(metric))
+  stopifnot(is.numeric(value) || is.na(value))
+  stopifnot(is.function(density_function))
+  stopifnot(is.integer(seg_start) || is.null(seg_start))
+  stopifnot(is.integer(seg_end) || is.null(seg_start))
+
+  # Creating object
+  x <- list(
+    par = par,
+    dist = distribution,
+    metric = metric,
+    value = value,
+    dens = density_function,
+    seg_start = seg_start,
+    seg_end = seg_end,
+    ...
+  )
+  class(x) <- "ModelFit"
+
+  return(x)
+}
+
+#' @export
+print.ModelFit = function(x, ...){
+
+  # Distribution/Metric/Observations
+  cat("ModelFit\n")
+  cat("Distribution: ", x$dist, "\n")
+  if(x$metric == "mle"){
+    cat("MLE: ", x$value, "\n\n")
+  } else {
+    cat("Metric: ", x$metric, "\n")
+    cat("Metric(obs, exp): ", x$value, "\n\n")
+  }
+
+  # Parameters
+  if(!is.null(x$par)){
+    cat("Fitted parameters:\n")
+
+    param_names <- names(x$par)
+    param_vals <- unlist(x$par)
+    param_vals <- as.character(formatC(param_vals, digits = 4))
+    spacing <- max(nchar(c(param_names, param_vals)))
+
+    param_names <- stringr::str_pad(param_names, width = spacing)
+    param_vals <- stringr::str_pad(param_vals, width = spacing)
+
+    cat(param_names, "\n")
+    cat(param_vals, "\n")
+  }
+
+  # Truncated Uniform
+  if(!is.null(x$seg_start) & !is.null(x$seg_end)){
```

Comment:
Use `&&` when comparing scalar logical values (`&` does vector based logical and).